src/app/asset-page/asset-page.component.ts

```diff
@@ -906,20 +906,20 @@ export class AssetPage implements OnInit, OnDestroy {
   /**
    * trackEvent
    * send captain's log events associated with asset
-   * @param asset
+   * @param assetId
    * @param eventType
    * @returns void
    */
-  private trackEvent(asset: Asset, eventType: string): void {
+  private trackEvent(assetId: string, eventType: string): void {
     const hasAccess = !this.showAccessDeniedModal
     const reasonForAuth = this.showAccessDeniedModal ? "not_authorized" : "license"
-    const abSegments = this._search.ab_segments.get(asset.id)
+    const abSegments = this._search.ab_segments.get(assetId)
 
     this._log.log({
       eventType: eventType,
       referring_requestid: this._search.latestSearchRequestId,
       ab_segments: abSegments ? [abSegments] : [],
-      item_id: asset.id,
```

Comment:
When the user does not have access to the asset, then `asset` will be undefined (because it gets loaded from the metadata service). Converted over to using the `this.assetIds` because we should always have that info regardless of access or not.